app/Http/Controllers/Auth/RegisterController.php

```diff
@@ -62,10 +66,19 @@ protected function validator(array $data)
      */
     protected function create(array $data)
     {
-        return User::create([
-            'name' => $data['name'],
+        $user = User::create([
+            'username' => $data['username'],
             'email' => $data['email'],
             'password' => bcrypt($data['password']),
+            'user_type' => (($data['user_type'] == 'author') ? 2 : 3)
         ]);
+
+        $user_details = UserDetail::create([
```

Comment:
Please use laravel way of saving related model ([check docs](https://laravel.com/docs/5.4/eloquent-relationships#inserting-and-updating-related-models)).
```
$user->userDetail()->create([ ... /*firstname & lastname only no more id*/ ]);